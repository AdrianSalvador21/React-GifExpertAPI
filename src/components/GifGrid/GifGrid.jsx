import React, {useState, useEffect} from 'react';
import GifGridItem from "../GifGridItem/GifGridItem";
import {getGifs} from "../../helpers/getGifs";
import {useFetchGifs} from "../../hooks/useFetchGifs";
import PropTypes from "prop-types";

const GifGrid = ({category}) => {
  const {data:images, loading} = useFetchGifs(category);

  // const [images, setImages] = useState([]);
  /* useEffect(() => {
    getGifs(category).then((imgs) => {
      setImages(imgs);
    });
  }, [category]); */
  return (
    <>
      <h3 className='animate__animated animate__fadeIn'>{category}</h3>
      {loading && <p>Cargando</p>}
      <div className="card-grid">
          {
            images.map((img) => {
              return (
                <GifGridItem key={img.id}
                             img={img}
                             {...img}
                />
              )
            })
          }
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
};

export default GifGrid;
