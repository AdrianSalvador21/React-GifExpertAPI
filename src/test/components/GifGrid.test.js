import '@testing-library/react';
import '@testing-library/jest-dom';
import {shallow} from "enzyme";
import React from "react";
import GifGrid from "../../components/GifGrid/GifGrid";
import {useFetchGifs} from "../../hooks/useFetchGifs";

jest.mock('../../hooks/useFetchGifs');

describe('TEST GifGridItem', () => {
  const category = 'Goku';

  test('should show component correctly', () => {
    useFetchGifs.mockReturnValue({
      data:[],
      loading: true
    });
    const wrapper = shallow(<GifGrid category={category}/>);

    expect(wrapper).toMatchSnapshot();
  });

  test('should show items when load images -> with useFetchGifs', () => {
    const gifs = [{id: '1', title: 'titulo', url: 'https://co.com'}];
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false
    });
    const wrapper = shallow(<GifGrid category={category}/>);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);

  });
});
