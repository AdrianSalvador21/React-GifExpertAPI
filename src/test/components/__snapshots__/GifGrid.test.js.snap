// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEST GifGridItem should show component correctly 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Goku
  </h3>
  <p>
    Cargando
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`TEST GifGridItem should show items when load images -> with useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Goku
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="1"
      img={
        Object {
          "id": "1",
          "title": "titulo",
          "url": "https://co.com",
        }
      }
      key="1"
      title="titulo"
      url="https://co.com"
    />
  </div>
</Fragment>
`;
